docker build -t placedock . fails due to missing ffi header

```
    Running setup.py install for cffi: started
    Running setup.py install for cffi: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /app/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-zd8qxx9v/cffi_95f3fac7e1764bd48c77d5bb0810be72/setup.py'"'"'; __file__='"'"'/tmp/pip-install-zd8qxx9v/cffi_95f3fac7e1764bd48c77d5bb0810be72/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-vz1j1rxk/install-record.txt --single-version-externally-managed --compile --install-headers /app/include/site/python3.10/cffi
         cwd: /tmp/pip-install-zd8qxx9v/cffi_95f3fac7e1764bd48c77d5bb0810be72/
    Complete output (56 lines):
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    Package 'libffi', required by 'virtual:world', not found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
FROM python:3-alpine as build-env
    Package 'libffi', required by 'virtual:world', not found
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.10
    creating build/lib.linux-x86_64-3.10/cffi
    copying cffi/cffi_opcode.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/lock.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/commontypes.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/api.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/setuptools_ext.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/cparser.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/vengine_cpy.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/model.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/verifier.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/pkgconfig.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/vengine_gen.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/error.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/recompiler.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/ffiplatform.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/__init__.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/backend_ctypes.py -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/_cffi_include.h -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/parse_c_type.h -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/_embedding.h -> build/lib.linux-x86_64-3.10/cffi
    copying cffi/_cffi_errors.h -> build/lib.linux-x86_64-3.10/cffi
    running build_ext
    building '_cffi_backend' extension
    creating build/temp.linux-x86_64-3.10
    creating build/temp.linux-x86_64-3.10/c
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/app/include -I/usr/local/include/python3.10 -c c/_cffi_backend.c -o build/temp.linux-x86_64-3.10/c/_cffi_backend.o
    c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
       15 | #include <ffi.h>
          |          ^~~~~~~
    compilation terminated.
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /app/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-zd8qxx9v/cffi_95f3fac7e1764bd48c77d5bb0810be72/setup.py'"'"'; __file__='"'"'/tmp/pip-install-zd8qxx9v/cffi_95f3fac7e1764bd48c77d5bb0810be72/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-vz1j1rxk/install-record.txt --single-version-externally-managed --compile --install-headers /app/include/site/python3.10/cffi Check the logs for full command output.
WARNING: You are using pip version 21.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/app/bin/python -m pip install --upgrade pip' command.

```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
 FROM python:3-alpine as build-env
 MAINTAINER Chris Dent <cdent@anticdent.org>
 
-RUN apk add --no-cache git gcc musl-dev linux-headers postgresql-dev pcre-dev
+RUN apk add --no-cache git gcc musl-dev linux-headers postgresql-dev pcre-dev libffi-dev
 
 RUN python -m venv /app
 

